### PR TITLE
Viet tag detail activity

### PR DESCRIPTION
**Pull request by viethphamvn**
Overview:
Refactor MapFragment for a better handling single or multiple tags
Changed Icons
Cleaned codes and added comments

### CODE REVIEW

File: app/src/main/java/edu/temple/templetag/fragments/MapFragment.java [hunk 1]
```diff
@@ -37,41 +37,56 @@
...
-//        args.putParcelable(ARG_PARAM2, currentUserLocation);
-//        fragment.setArguments(args);
+        Bundle args = new Bundle();
+        args.putParcelableArrayList(ARG_PARAM1, tags);
+        args.putParcelable(ARG_PARAM2, currentLocation);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    public static MapFragment newInstance(Tag tag, Location currentLocation) {
```
Comment (dilloncoffman) [inline app/src/main/java/edu/temple/templetag/fragments/MapFragment.java:71]: Are we passing one tag into the MapFragment or should it be an ArrayList<Tag>?

File: app/src/main/java/edu/temple/templetag/fragments/MapFragment.java [hunk 2]
```diff
@@ -37,41 +37,56 @@
 import edu.temple.templetag.HomeActivity;
 import edu.temple.templetag.R;
 import edu.temple.templetag.Tag;
+import edu.temple.templetag.tools.DistanceCalculator;
 
 public class MapFragment extends Fragment implements OnMapReadyCallback {
     private static final String ARG_PARAM1 = "tags";
```
Comment (dilloncoffman) [inline app/src/main/java/edu/temple/templetag/fragments/MapFragment.java]: Could we update the names for these constant keys from ARG_PARAM# to be like TAGS and TAG so things are more readable below?

File: app/src/main/java/edu/temple/templetag/fragments/MapFragment.java (hunk 1 above)
Comment (viethphamvn) [reply app/src/main/java/edu/temple/templetag/fragments/MapFragment.java:71]: From HomeActivity we will pass an ArrayList but in tagDetailActivity, we only need to pass one tag

Comment (dilloncoffman) [inline app/src/main/java/edu/temple/templetag/fragments/MapFragment.java:70]: What are the two newInstance methods for in MapFragment? How are they used?

File: app/src/main/java/edu/temple/templetag/fragments/MapFragment.java [hunk 3]
```diff
@@ -37,41 +37,56 @@
...
-    GoogleMap googleMap;
+    public GoogleMap googleMap;
 
     public MapFragment() {
         // Required empty public constructor
     }
 
 
-    public static MapFragment newInstance() {
+    public static MapFragment newInstance(ArrayList<Tag> tags, Location currentLocation) {
```
Comment (dilloncoffman) [inline app/src/main/java/edu/temple/templetag/fragments/MapFragment.java:61]: What are the two newInstance methods for in MapFragment? How are they used?

File: app/src/main/java/edu/temple/templetag/fragments/MapFragment.java [hunk 4]
```diff
@@ -37,41 +37,56 @@
...
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-//        if (getArguments() != null) {
-//            tags = getArguments().getParcelableArrayList(ARG_PARAM1);
-//            currentUserLocation = getArguments().getParcelable(ARG_PARAM2);
-//        }
+        if (getArguments() != null) {
+            tags = getArguments().getParcelableArrayList(ARG_PARAM1);
+            if (tags == null){
```
Comment (dilloncoffman) [inline app/src/main/java/edu/temple/templetag/fragments/MapFragment.java:84]: Refactor ARG_PARAM naming for readability

File: app/src/main/java/edu/temple/templetag/fragments/MapFragment.java [hunk 5]
```diff
@@ -106,43 +119,42 @@ public void displayMarkers(ArrayList<Tag> Tags, Location currentLocation){
...
-        double r = 3956;
+    public void updateNewTags(ArrayList<Tag> tags, Location currentLocation){
+        this.tags = tags;
+        this.currentLocation = currentLocation;
+        displayMarkers(this.tags);
+    }
 
-        // calculate the result
-        return(c * r);
+    public void updateNewATag(Tag tag, Location currentLocation){
```
Comment (dilloncoffman) [inline app/src/main/java/edu/temple/templetag/fragments/MapFragment.java]: Could this be named updateNewTagLocation()?

File: app/src/main/java/edu/temple/templetag/fragments/MapFragment.java [hunk 6]
```diff
@@ -106,43 +119,42 @@ public void displayMarkers(ArrayList<Tag> Tags, Location currentLocation){
...
 
-        double c = 2 * Math.asin(Math.sqrt(a));
+            focusOn(currentLocation);
+        }
+    }
 
-        // Radius of earth in kilometers. Use 3956
-        // for miles
-        double r = 3956;
+    public void updateNewTags(ArrayList<Tag> tags, Location currentLocation){
```
Comment (dilloncoffman) [inline app/src/main/java/edu/temple/templetag/fragments/MapFragment.java]: Could this be named updateNewTagsLocations?

File: app/src/main/java/edu/temple/templetag/TagDetailActivity.java [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+        tagDesc = findViewById(R.id.tag_description);
+
+        tagLocationName.setText(mTag.getmTagLocationName());
+        tagCreatedBy.setText("created by " + mTag.getmTagCreatedBy());
+        tagUpVote.setText(mTag.getmTagUpvoteCount() + " Upvotes");
+        tagDownVote.setText(mTag.getmTagDownvoteCount() + " Downvotes");
+        tagPop.setText(mTag.getmTagPopularity() + " people are talking about this event");
+        tagDesc.setText(mTag.getmTagDescription());
+
+        closeBtn = findViewById(R.id.btn_delete);
```
Comment (dilloncoffman) [inline app/src/main/java/edu/temple/templetag/TagDetailActivity.java:56]: Is this button to delete an individual tag? I assume a user should only be able to delete a tag if they're the one that created it.

**Review by dilloncoffman — CHANGES_REQUESTED** [review]
Left some comments for your review but solid work! You've really been killing it man 🔥

File: app/src/main/java/edu/temple/templetag/fragments/MapFragment.java (hunk 3 above)
Comment (viethphamvn) [reply app/src/main/java/edu/temple/templetag/fragments/MapFragment.java:61]: The MapFragment in TagDetailActivity only needs one tag whereas MapFragment in HomeActivity needs an ArrayList of Tag.

**Pull request closed without merging**